### ISSUE TITLE
Keep missing attributes

### ISSUE THREAD

**Issue by podsvirov** [asciidoc-py/asciidoc-py#200]
Some have trouble using it in a [Pelican](https://github.com/getpelican/pelican-plugins/issues/1051).

Sample code `test.adoc`:
```adoc
{attach}file.txt
```

Good output for `asciidoctor --no-header-footer test.adoc`:
```html
<div class="paragraph">
<p>{attach}file.txt</p>
</div>
```

Bad output for `asciidoc --no-header-footer test.adoc`:
```html
<div class="paragraph"><p></p></div>
```

How to save or render attributes with curly braces?
Setting the attribute doesn't help `asciidoc --no-header-footer -a attach={attach} test.adoc`:
```html
<div class="paragraph"><p></p></div>
```
*UPDATE*:
But `asciidoc --no-header-footer -a attach={} test.adoc` produce:
```html
<div class="paragraph"><p>{}file.txt</p></div>
```

**Comment by elextr** [comment]
Please note the note in this section of the Asciidoctor manual https://docs.asciidoctor.org/asciidoc/latest/attributes/unresolved-references/#missing.  The two processors behave differently.

**Comment by podsvirov** [comment]
@elextr thanks, but unfortunately `asciidoc` don't support same `attribute-missing` attribute to control missing attributes.

**Comment by elextr** [comment]
Correct, as the note says the Asciidoc behaviour is part of the implementation and the `missing-attribute` is an Asciidoctor addition.  

Since as the note says the feature of dropping lines is used in the Asciidoc implementation so it could be difficult to accommodate global settings like `missing-attribute` since it would possibly break the implementation if the behaviour changed.

**Comment by podsvirov** [comment]
For `Pelican` I need to skip `{attach}` at `asciidoc` rendering step. What can I do to get this behavior?

**Comment by elextr** [comment]
https://asciidoc-py.github.io/userguide.html#_attribute_references

First item in the list.

**Comment by podsvirov** [comment]
Yoohoo! After change code sample, I've similar output from both tools. Thanks!

The `test.adoc`:
```adoc
\{attach}file.txt
```
For `asciidoctor --no-header-footer test.adoc`:
```html
<div class="paragraph">
<p>{attach}file.txt</p>
</div>
```

For `asciidoc --no-header-footer test.adoc`:
```html
<div class="paragraph"><p>{attach}file.txt</p></div>
```

But I can't use this for links...

The `link.adoc`:
```adoc
link:\{attach}file.txt[file]
```

For `asciidoctor --no-header-footer link.adoc`:
```html
<div class="paragraph">
<p><a href="{attach}file.txt">file</a></p>
</div>
```

For `asciidoc --no-header-footer link.adoc`:
```html
<div class="paragraph"><p></p></div>
```

You've already helped me, but do you have any more hints?

**Comment by MasterOdin** [comment]
This is a bug in the processor, where escaped attributes do not work at all within macro definitions. I've opened #205 which fixes this particular case, and an explanation of why it didn't work, but I'm not necessarily thrilled with the solution as such, and I fear I've broken some edge case of macro expansion in trying to fix this.

@elextr Are you aware of any particular gnarly macros that might trip up those changes that are not currently defined within the test suite?

**Comment by elextr** [comment]
Not really, my comment above was because I believe that parts of the config and backend config depend on the dropping of lines with missing attributes.  I hadn't considered macro targets.

Does escaping the escape so it gets to the second phase work as a fallback if there are problems with the fix?
```
link:\\{attach}file.txt[file]
```

**Comment by MasterOdin** [comment]
Unfortunately not. While the additional escape gets past the inner parse of `{attach}file.txt`, the final parse of `<a href="{attach}foo.jpg">{attach}foo.jpg</a>` fails. I've handled the basics of this, but:

> believe that parts of the config and backend config depend on the dropping of lines with missing attributes

Yeah, that is the devil in the detail here, and why asciidoc.py will probably never be able to support `missing-attribute`. However, playing with this more, I think I found a bit more robust solution than my initial attempt. It passes the test suite at least, and does not break on any of the edge cases that #205 had. I'll leave it open for a bit as I think through possible ramifications and if you have any ideas on what could go wrong with the approach. Granted, this is definitely showing some of the limitations of the regex based parser.
